File: VB/WindowsApplication1/CustomEditor/MyButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Popup

Namespace WindowsApplication1

	Public Class MyPopupContainerEdit
		Inherits PopupContainerEdit
		Shared Sub New()
			RepositoryItemMyPopupContainerEdit.Register()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyPopupContainerEdit.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyPopupContainerEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyPopupContainerEdit)
			End Get
		End Property

		Protected Overrides Function CreatePopupForm() As DevExpress.XtraEditors.Popup.PopupBaseForm
			Return New MyPopupContainerForm(Me)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/CustomEditor/MyButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Popup;

namespace WindowsApplication1
{
  
    public class MyPopupContainerEdit : PopupContainerEdit
    {
        static MyPopupContainerEdit()
        {
            RepositoryItemMyPopupContainerEdit.Register();
        }
     
        public override string EditorTypeName
        {
            get { return RepositoryItemMyPopupContainerEdit.EditorName; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyPopupContainerEdit Properties
        {
            get { return base.Properties as RepositoryItemMyPopupContainerEdit; }
        }

        protected override DevExpress.XtraEditors.Popup.PopupBaseForm CreatePopupForm()
        {
            return new MyPopupContainerForm(this);
        }
    }
}